tools/openms/MSstatsConverter.xml

```diff
@@ -24,7 +24,8 @@ mkdir out &&
 #if $adv_opts_cond.adv_opts_selector=='advanced':
   #if $adv_opts_cond.reannotate_filenames:
     mkdir adv_opts_cond.reannotate_filenames &&
-    ${ ' '.join(["ln -s '%s' 'adv_opts_cond.reannotate_filenames/%s.%s' &&" % (_, re.sub('[^\w\-_]', '_', _.element_identifier), $gxy2omsext(_.ext)) for _ in $adv_opts_cond.reannotate_filenames if _]) }
+    mkdir ${' '.join(["'adv_opts_cond.reannotate_filenames/%s'" % (i) for i, f in enumerate($adv_opts_cond.reannotate_filenames) if f])} && 
+    ${' '.join(["ln -s '%s' 'adv_opts_cond.reannotate_filenames/%s/%s.%s' && " % (f, i, re.sub('[^\w\-_]', '_', f.element_identifier), $gxy2omsext(f.ext)) for i, f in enumerate($adv_opts_cond.reannotate_filenames) if f])}
```

Comment:
@bernt-matthias a very general question, should we maybe have `f.save_element_identifier` that is doing this inside Galaxy, s that we can avoid all this sed/replace stuff in the wrappers?

Reply:
Yeah! This is used in so many tools. Started almost 2 years ago with this https://github.com/galaxyproject/galaxy/pull/7937 .. somehow got distracted. We should try to get this in 21.05.

Reply:
Oh gosh, complete forget about it ... :100: